src/Panel/Panel.php

```diff
@@ -450,19 +450,18 @@ public static function routesForViews(string $areaId, array $area): array
      */
     public static function setLanguage(): ?string
     {
-        $kirby = kirby();
+        $kirby           = kirby();
+        $session         = $kirby->session();
```

Comment:
I found a new clue regarding the issue. Maybe this will give you an idea. Actually all the error is related to this line. Just executing this line in the method is enough (outside the if condition). When I remove this line, the tests fail. What happens when the `$kirby->session()` method is called?

Reply:
It's not a solution but you can see how it works. `$session = $kirby->session();` must be outside the if condition in this case. So why? I don't know 🙂

Reply:
How weird is that? @lukasbestle do you have any idea?

Reply:
Pinging @lukasbestle to get any idea 🙂

Reply:
I tracked it down to the `tmp/site/sessions` directory being created by `$kirby->session()`. If I put the following in [this line](https://github.com/getkirby/kirby/blob/develop/tests/Panel/Areas/AreaTestCase.php#L179), then everything works again:

```php
Dir::make($this->tmp . '/site/sessions', true);
```

No idea what in the area tests depends on an empty session directory though. 🤯

Reply:
@lukasbestle I guess found the issue.

Follow `$system->isOk()` in following line. This is about writable `sessions` directory. Otherwise redirecting `installation` path and test fail.

https://github.com/getkirby/kirby/blob/fix/panel-setting-language/src/Panel/Panel.php#L65

Reply:
Ah, of course! I think we should create the session dir in `$system->init()` like we do with `accounts`.

Reply:
✅